### PR TITLE
Implemented a timer to the game !!!

### PR DESCRIPTION
**Pull request by christos-kouros**
Hi , my name is Chris and i am a computer science student . I was searching for random chess games but yours really stood out. I took some time and implemented a timer to the game. The user can change its status to on/off or even change the timer itself to make the game more or less challenging !!

I am looking forward to your answer and maybe the merge of my commits to your original code.
Thank you in advance.
Kouros Christos

### CODE REVIEW

**Comment by GervinFung** [comment]
Hey Chris, I really appreciate it, thank you for your compliment and taking
your time to implement a timer function, it means a lot!!. I did take a
good look at your program, the outcome of the program is great. However, I noticed you used deprecated functions as well such as
suspend() and resume(), which is not recommended, I am not an expert but I
do not like to use deprecated functions or class, which is why I replaced
Observer with PropertyChangeSupport and hence the reason I did not merge
and I apologize for that. Lastly, I hope we can collaborate on some
projects in the future!!! Do add me on discord if you have an account, my
ID is PoolOfDeath20#8909

**Comment by christos-kouros** [comment]
Hi again ! Thanks for taking some time to review my pull request and answering ! You are absolutely right about the use of the deprecated functions , i should have avoided them in the first place . Working together on some future projects sounds like fun , so i am going to add you on discord !

File: src/chess/gui/Table.java [hunk 1]
```diff
@@ -141,9 +151,34 @@ private Table() {
...
+        blackPlayer = new Thread(b);
+
+        whitePlayerSeconds = time;
+        blackPlayerSeconds = time;
+
+        whitePlayer.start();
+        blackPlayer.start();
+        blackPlayer.suspend();
+
+        this.getGameSetup().whitePlayer = whitePlayer;
```
Comment (aimiebell) [inline src/chess/gui/Table.java:171]: I detect that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [SA: Self assignment of value (SA_SELF_ASSIGNMENT)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#sa-self-assignment-of-field-sa-field-self-assignment).
This method contains a self assignment of a field; e.g.
Such assignments are useless, and may indicate a logic error or typo.

Comment (GervinFung) [reply src/chess/gui/Table.java:171]: Ya I did noticed that too but forgot to point out as well, thank you aimieball for pointing that out

**Comment by GervinFung** [comment]
Hi aimiebell, thank you for pointing this out as I had noticed it as well
but forgot to point it out. Thank you very much !!!

**Pull request closed without merging**